src/utils/BeameUtils.js

```diff
@@ -88,6 +87,77 @@ function findInTree(node, predicate, limit) {
 }
 //endregion
 
+//region Background Jobs
+/**
+ * @typedef {Object} BackgroundJob
+ * @property {Object} handle => set interval handle
+ * @property {Number} interval => interval to run the job
+ * @property {Number} called => times the job was called
+ * @property {Number} failed => times the job failed
+ * @property {Boolean} running => will be true while the process is running
+ * @property {* | undefined} lastResult => result from the last run
+ */
+/**
+ * @type {{name: BackgroundJob}}
+ */
+const backgroundJobs = {};
+/**
+ * Starts a background job that will run every interval
+ * @param {String} name
+ * @param {Function} func
+ * @param {Number} interval (in ms)
+ * @returns {boolean}
+ */
+function startBackgroundJob(name, func, interval) {
+	if(backgroundJobs[name]) {
+		logger.warn(`Background job '${name}' is already running, skipping....`);
+		return false;
+	}
+
+	async function jobCall() {
+		backgroundJobs[name].called++;
+		backgroundJobs[name].running = true;
+		try {
+			backgroundJobs[name].lastResult = await func();
+		}
+		catch(e) {
+			backgroundJobs[name].lastResult = e.message;
```

Comment:
Information loss:
1. The exception is not printed
1. The stack trace is lost forever
1. There is no way to programmatically distinguish between `lastResult` which is result of `await func()` and one that comes from `e.message`

Reply:
fixed